components/scream/src/physics/cld_fraction/atmosphere_cld_fraction.cpp

```diff
@@ -0,0 +1,135 @@
+{
+  using namespace ekat::units;
+
+  // The units of mixing ratio Q are technically non-dimensional.
+  // Nevertheless, for output reasons, we like to see 'kg/kg'.
+  auto Q = kg/kg;
+  Q.set_string("kg/kg");
+  Units nondim(0,0,0,0,0,0,0);
+
+  const auto& grid_name = m_cld_fraction_params.get<std::string>("Grid");
+  auto grid  = grids_manager->get_grid(grid_name);
+  m_num_cols = grid->get_num_local_dofs(); // Number of columns on this rank
+  m_num_levs = grid->get_num_vertical_levels();  // Number of levels per column
+
+  // Define the different field layouts that will be used for this process
+  using namespace ShortFieldTagsNames;
+
+  // Layout for 3D (2d horiz X 1d vertical) variable defined at mid-level and interfaces 
+  FieldLayout scalar3d_layout_mid { {COL,LEV}, {m_num_cols,m_num_levs} };
+
+  // Set of fields used strictly as input
+  m_required_fields.emplace("qi",   scalar3d_layout_mid, Q,      grid_name);
+  m_required_fields.emplace("alst", scalar3d_layout_mid, nondim, grid_name);
+
+  // Set of fields used strictly as output
+  m_computed_fields.emplace("ast",   scalar3d_layout_mid, nondim, grid_name);
+  m_computed_fields.emplace("aist",  scalar3d_layout_mid, nondim, grid_name);
+
+  // Set of fields used as input and output
+  // - There are no fields used as both input and output.
+
+}
+
+// =========================================================================================
+void CldFraction::initialize_impl (const util::TimeStamp& t0)
+{
+}
+
+// =========================================================================================
+void CldFraction::run_impl (const Real dt)
+{
+  
+  // Calculate ice cloud fraction and total cloud fraction given the liquid cloud fraction
+  // and the ice mass mixing ratio. 
+  auto qi   = m_cld_fraction_fields_in["qi"].get_reshaped_view<const Pack**>();
+  auto liq_cld_frac = m_cld_fraction_fields_in["alst"].get_reshaped_view<const Pack**>();
+  auto ice_cld_frac = m_cld_fraction_fields_out["aist"].get_reshaped_view<Pack**>();
+  auto tot_cld_frac  = m_cld_fraction_fields_out["ast"].get_reshaped_view<Pack**>();
+
+  CldFractionFunc::main(m_num_cols,m_num_levs,qi,liq_cld_frac,ice_cld_frac,tot_cld_frac);
+
+  // Get a copy of the current timestamp (at the beginning of the step) and
+  // advance it,
+  auto ts = timestamp();
+  ts += dt;
+  for (auto& f : m_cld_fraction_fields_out) {
+    f.second.get_header().get_tracking().update_time_stamp(ts);
+  }
+
+}
+
+// =========================================================================================
+void CldFraction::finalize_impl()
+{
+  // Do nothing
+}
+
+// =========================================================================================
+void CldFraction::register_fields (FieldRepository<Real>& field_repo) const {
+  std::set<ci_string> q_names =
+    { "qi" };
+
+  for (const auto& fid : m_required_fields) {
+    const auto& name = fid.name();
+    if (q_names.count(name)>0) {
```

Comment:
Since there's only one entry in `q_name`, you could do something more readabe:

```
if (name=="qi")
```
and get rid of `q_names`.